data.py

```diff
@@ -27,48 +27,50 @@ def make_masks(src, tgt, device, pad=0):
     src_mask = (src != pad).unsqueeze(-2)
     return src_mask, tgt_mask
 
-
-class IMDBDataset(torch.utils.data.Dataset):
-    ''' Generate sentences with gt target for encoding '''
-
-    def __init__(self, imdb_dataset, tokenizer, max_len=MAX_LEN):
-        self.data = self._process_data(imdb_dataset)
-        self.tokenizer = tokenizer
-        self.max_len = max_len
-
-    def _process_data(self, imdb_dataset):
-        ''' From raw data to processed sentences '''
-        data_list = []
-        for row in imdb_dataset.rows:
-            text = BeautifulSoup(row["text"], 'html.parser').get_text()
-            target = True if row['sentiment'] == 'pos' else False
-
-            sents = sent_tokenize(text)
-            for sent in sents:
-                data_list.append({'text': sent, 'label': target})
-        return data_list
-
-    def __getitem__(self, ind):
-        item = self.data[ind]
-
-        data = item['text']
-        target = item['label']
-
-        indexed_tokens = self.tokenizer.encode(data)
-        curr_len = len(indexed_tokens)
-
-        if curr_len < self.max_len:
-            indexed_tokens = indexed_tokens + np.zeros((self.max_len - curr_len)).tolist()
-
-        elif curr_len > self.max_len:
-            indexed_tokens = indexed_tokens[:self.max_len]
-
-        else:
-            pass
-
-        src = torch.tensor(indexed_tokens).long()
-        label = torch.tensor(int(target))
-        return src, label
-
-    def __len__(self):
-        return len(self.data)
\ No newline at end of file
+def load_dataset(data_source, fix_length, device, batch_size_train, batch_size_test):
+    """
+    tokenizer : Breaks sentences into a list of words. If sequential=False, no tokenization is applied
+    Field : A class that stores information about the way of preprocessing
+    fix_length : An important property of TorchText is that we can let the input to be variable length, and TorchText will
+                 dynamically pad each sequence to the longest sequence in that "batch". But here we are using fi_length which
+                 will pad each sequence to have a fix length of 200.
+
+    build_vocab : It will first make a vocabulary or dictionary mapping all the unique words present in the train_data to an
+                  idx and then after it will use GloVe word embedding to map the index to the corresponding word embedding.
+
+    vocab.vectors : This returns a torch tensor of shape (vocab_size x embedding_dim) containing the pre-trained word embeddings.
+    BucketIterator : Defines an iterator that batches examples of similar lengths together to minimize the amount of padding needed.
+
+    """
+    # define tokenizer
+    en = English()
+    def tokenize_spacy_with_html_parsing(sentence):
```

Comment:
does it accept a sentence, or a few sentences? I think you can't really divide it to sentences unless you strip html first.